tests/dataset_test.py

```diff
@@ -0,0 +1,141 @@
+from qri import dataset
+from qri import loader
+import mock_loader
+import unittest
+
+# TODO - move these to a better place?
+LIST_OBJ = {
+    'username': 'me',
+    'profileID': 'Qm.profileID',
+    'name': 'first_dataset',
+    'path': '/ipfs/Qm.path',
+    'bodySize': 1,
+    'bodyRows': 2,
+    'numErrors': 3,
+    'bodyFormat': 'json',
+    'commitTime': '2020-08-24T20:00:32.171549113Z',
+    'fsiPath': '/home/my_user/datasets/first_dataset'
+}
+
+GET_OBJ = {
+    "bodyPath": "/ipfs/Qm.bodyPath",
+    "commit": {
+        "author": {"id": "Qm.commit.author.id"},
+        "message": "commit.message",
+        "timestamp": "2020-01-02T03:04:05.124963898Z",
+    },
+    "meta": {
+        "description": "meta.description",
+        "keywords": ["meta.keyword1", "meta.keyword2"],
+    },
+    "name": "first_dataset",
+    "path": "/ipfs/Qm.path",
+    "peername": "me",
+    "previousPath": "/ipfs/Qm.previousPath",
+    "readme": {
+        "scriptBytes": "VEVTVCBSRUFETUUK",
+    },
+    "structure": {
+        "checksum": "Qm.structure.checksum",
+        "entries": 3,
+        "format": "csv",
+        "schema": {
+            "items": {
+                "items": [
+                    {"title": "field1", "type": "string"},
+                    {"title": "field2", "type": "integer"},
+                ],
+                "type": "array",
+            },
+            "type": "array",
+        },
+    },
+}
+
+
+class DatasetTests(unittest.TestCase):
+    def test_init_from_list(self):
+        loader.set_instance(mock_loader.DatasetMockLoader())
```

Comment:
This isn't needed, the test doesn't do any loading, please remove.

Reply:
It asserts that `list_dataset_objects`, `get_dataset_object` and `load_body` never get called.

Reply:
What if we have the arguments required so that it's explicit?

```
loader.set_instance(mock_loader.DatasetMockLoader(
  list_response=None,
  get_responses={},
  get_body_responses={}
))
```

---

Alternately I was considering, instead of throwing an error when these functions are called without an override, we could also assert at the end of the test that the functions were not called:

```
dml = mock_loader.DatasetMockLoader()
loader.set_instance(dml)
...
self.assertEqual(dml.list_dataset_objects_calls, 0)
self.assertEqual(dml.get_dataset_object_calls, 0)
self.assertEqual(dml.load_body_calls, 0)
```

The problem is that I'm not sure what I'd do instead of throwing an error if they are called unexpectedly. We don't want to call the qri cli tool.

Reply:
Or maybe:
```
loader.set_instance(mock_loader.DatasetMockLoader(expected_calls={
    'get_responses': {
          'peer/first_dataset': GET_OBJ
    }
}))
```
vs

```
loader.set_instance(mock_loader.DatasetMockLoader(expected_calls={}))
```

Reply:
I see, I didn't understand the intention. This is fine for now, I'll follow up with another change to make it more explicit.

Reply:
You know what - my apologies, I wasn't thinking carefully when I responded. I assumed that you were responding to the lack of parameters to `DatasetMockLoader()`. I realize now that you're saying the test doesn't do any loading, and really there's no reason you'd expect it would. I guess I added this out of an abundance of caution. `test_dont_populate_twice` on the other hand I do explicitly want to make sure that it doesn't do any loading, and it plausibly might. For `test_init_from_get`, I'm not explicitly expecting loading, but I call `ds.meta` just as in `test_dont_populate_twice` so practically speaking it's just as likely to accidentally call a loading function.

So, take it as an abundance of caution. Up to you if you want to remove it.